### PR TITLE
[Feature] Brax envs

### PR DESCRIPTION
**Pull request by yingchenlin**
## Description

Draft PR of Brax envs

### CODE REVIEW

File: test/test_libs.py [hunk 1]
```diff
@@ -425,6 +427,87 @@ def test_jumanji_consistency(self, envname, batch_size):
                 )
 
 
+@pytest.mark.skipif(not _has_brax, reason="brax not installed")
+@pytest.mark.parametrize("envname", ["fast"])
+class TestBrax:
```
Comment (vmoens) [inline test/test_libs.py:432]: None of the tests was run
We should create a dedicated pipeline (see how we did for jumanji in .circleci)

File: torchrl/envs/libs/brax.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+import torch
+from tensordict.tensordict import TensorDict, TensorDictBase
+
+from torchrl.data import (
+    CompositeSpec,
+    NdBoundedTensorSpec,
+    NdUnboundedContinuousTensorSpec,
+)
+from torchrl.envs.common import _EnvWrapper
+from torchrl.envs.libs.jumanji import _torchrl_data_to_spec_transform
```
Comment (vmoens) [inline torchrl/envs/libs/brax.py:12]: we should definitely not import stuff from lib to lib.
Here I would put `_torchrl_data_to_spec_transform` in a utils file, jax_utils (and rename it to a less generic name).

**Review by vmoens — COMMENTED** [review]
Let's set up the test pipeline, I'll give it a second look once it's done

**Comment by vmoens** [comment]
The tests are passing! Just a few edits and we're good to go!

File: torchrl/envs/libs/brax.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        _tensor_to_ndarray,
+        _tensordict_to_object,
+        _tree_flatten,
+        _tree_reshape,
+    )
+
+    _has_brax = True
+except ImportError as err:
+    _has_brax = False
+    IMPORT_ERR = str(err)
```
[the remark this answers is not in the document]
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py:30]: done

File: torchrl/envs/libs/brax.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+
+class BraxWrapper(_EnvWrapper):
+    """Google Brax environment wrapper.
+
+    Examples:
+        >>> env = brax.envs.get_environment("ant")
+        >>> env = BraxWrapper(env)
+        >>> td = env.rand_step()
+        >>> print(td)
```
[the remark this answers is not in the document]
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py:40]: done

File: torchrl/envs/libs/brax.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+        >>> print(env.available_envs)
```
[the remark this answers is not in the document]
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py:41]: done

File: torchrl/envs/libs/brax.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+    def _make_specs(self, env: "brax.envs.env.Env") -> None:  # noqa: F821
+        self._input_spec = CompositeSpec(
+            action=NdBoundedTensorSpec(
+                minimum=-1, maximum=1, shape=(env.action_size,), device=self.device
+            )
+        )
+        self._reward_spec = NdUnboundedContinuousTensorSpec(
+            shape=(), device=self.device
+        )
+        self._observation_spec = CompositeSpec(
```
[the remark this answers is not in the document]
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py:100]: I think it is. There's no doc yet, but it's directly used by Brax policies.
https://github.com/google/brax/blob/main/brax/training/agents/sac/networks.py#L54

File: torchrl/envs/libs/brax.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+        # build result
+        tensordict_out = TensorDict(
+            source={
+                "observation": state.get("obs"),
+                "reward": state.get("reward"),
+                "done": state.get("done").bool(),
+                "state": state,
+            },
+            batch_size=self.batch_size,
+            device=self.device,
```
[the remark this answers is not in the document]
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py:148]: done

File: torchrl/envs/libs/brax.py [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+        # build result
+        tensordict_out = TensorDict(
+            source={
+                "observation": next_state.get("obs"),
+                "reward": next_state.get("reward"),
+                "done": next_state.get("done").bool(),
+                "state": next_state,
+            },
+            batch_size=self.batch_size,
+            device=self.device,
```
[the remark this answers is not in the document]
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py:178]: done

File: torchrl/envs/libs/brax.py (hunk 2 above)
Comment (yingchenlin) [reply torchrl/envs/libs/brax.py]: done

**Review by vmoens — APPROVED** [review]
LGTM 
Let's wait for the tests to pass and merge this!

**Pull request merged**